### PR TITLE
Implements $setEquals operator

### PR DESCRIPTION
**Pull request by lucasrcezimbra**

### CODE REVIEW

File: tests/test__collection_api.py [hunk 1]
```diff
@@ -3454,11 +3454,6 @@ def test__aggregate_not_implemented(self):
                 {'$project': {'a': {'$concatArrays': [[0, 1], [2, 3]]}}},
             ])
 
-        with self.assertRaises(NotImplementedError):
-            self.db.collection.aggregate([
-                {'$project': {'a': {'$setEquals': [[2], [1, 2, 3]]}}},
```
Comment (pcorpet) [inline tests/test__collection_api.py:3537]: Please keep this one but with `setIntersection` to keep code coverage high.

File: tests/test__collection_api.py [hunk 2]
```diff
@@ -4682,6 +4677,32 @@ def test__set_union(self):
         }]
         self.assertEqual(expect, list(actual))
 
+    def test__set_equals(self):
+        collection = self.db.collection
+        collection.insert_many([
+            {'array': ['one', 'three']},
+        ])
+        actual = collection.aggregate([{'$project': {
```
Comment (pcorpet) [inline tests/test__collection_api.py:4800]: Please also add an equivalent case in test__mongomock to make sure the behavior is the same than the one of pymongo.

**Review by pcorpet — CHANGES_REQUESTED** [review]
Thanks a lot for implementing the feature.

Let's make sure that all CI checks are successful.

File: tests/test__collection_api.py (hunk 1 above)
Comment (lucasrcezimbra) [reply tests/test__collection_api.py:3537]: Done!

File: tests/test__collection_api.py (hunk 2 above)
Comment (lucasrcezimbra) [reply tests/test__collection_api.py:4800]: Done!

**Review by pcorpet — APPROVED** [review]

**Pull request merged**